adoc/chapters/programming_interface.adoc

```diff
@@ -8195,7 +8193,9 @@ local_accessor()
   * All size queries return [code]#0#.
   * The return values of [code]#get_pointer()# and [code]#get_multi_ptr()# are
     unspecified.
-  * Trying to access the underlying memory is undefined behavior.
+  * A default constructed local accessor can be passed to a
+    <<sycl-kernel-function>>, but attempting to access data elements from it
+    produces undefined behavior.
```

Comment:
We discussed the question about `nullptr` already in #232, see [this comment](https://github.com/KhronosGroup/SYCL-Docs/pull/232#discussion_r839651498).

My opinion is pretty much the same as before.  I think we should treat all zero-sized accessors the same in this regard (either say they all return `nullptr` or say they all return an unspecified value).  It is not hard to guarantee the `nullptr`, but it does require extra logic in the implementation.  For example, when constructing a ranged accessor, the implementation would need a special check to see if the range was zero.  Guaranteeing a `nullptr` does not seem that useful to me.  Applications still need to check that the index is within range of the accessor.  If applications follow this rule, then they should never attempt to dereference the pointer for a zero-sized accessor.  If applications do want to check for this special case, it seems easy enough to call `accessor::empty`.

Reply:
Sorry, I missed that discussion, I think that's a fair point, though I think there is still a use case for this, though admittedly pretty niche. If you had a kernel where an accessor is conditionally used, being left as default constructed if it's not, and when it is used each work-item only accesses its own index, and it's guaranteed to be the same size as the nd-range, then you may want to check if the accessor is zero-sized, without having to check the size (and potentially removing the implementation's ability to optimize away that kernel argument).

Though saying this I think while `size()` does require an argument to be passed to the kernel to provide this value, `empty` could be implemented based on whether the pointer is `nullptr`. So I think this is a problem that can be handled by the implementation, and so doesn't need to be in the specification.

Reply:
Rather than holding up this PR, lets create an issue to discuss whether a zero-sized accessor should be required to return `nullptr` from `get_pointer`.  I think this is an orthogonal discussion from the clarification I'm making in this PR.